src/Sulu/Bundle/ContentBundle/Controller/NodeController.php

```diff
@@ -246,37 +249,40 @@ private function getTreeContent(
      * @param string  $uuid
      *
      * @return \Symfony\Component\HttpFoundation\Response
-     *
-     * @deprecated this will be removed when the content-repository is able to solve all requirements.
      */
     private function getSingleNode(Request $request, $uuid)
     {
         $language = $this->getLanguage($request);
-        $webspace = $this->getWebspace($request, false);
         $breadcrumb = $this->getBooleanRequestParameter($request, 'breadcrumb', false, false);
         $complete = $this->getBooleanRequestParameter($request, 'complete', false, true);
         $ghostContent = $this->getBooleanRequestParameter($request, 'ghost-content', false, false);
 
         $view = $this->responseGetById(
             $uuid,
-            function ($id) use ($language, $webspace, $breadcrumb, $complete, $ghostContent) {
+            function ($id) use ($language, $ghostContent) {
                 try {
-                    return $this->getRepository()->getNode(
-                        $id,
-                        $webspace,
-                        $language,
-                        $breadcrumb,
-                        $complete,
-                        $ghostContent
-                    );
+                    return $this->getDocumentManager()->find($id, $language, [
+                        'load_ghost_content' => $ghostContent,
+                    ]);
                 } catch (DocumentNotFoundException $ex) {
                     return;
                 }
             }
         );
 
+        $groups = [];
+        if (!$complete) {
+            $groups[] = 'smallPage';
+        } else {
+            $groups[] = 'defaultPage';
+        }
+
+        if ($breadcrumb) {
+            $groups[] = 'breadcrumbPage';
+        }
```

Comment:
I would prefer the current format, because the contexts are already separated by `,` in the serializer xml configuration. Then you would end up writing something like `page.small,page.default` in there, which is not as easy to read IMO.